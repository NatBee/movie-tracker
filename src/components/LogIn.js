import React, { Component } from 'react';
import { redirect, withRouter } from 'react-router-dom';
import { setUser } from '../actions/index';
import * as actions from '../actions/index';
import { userSignIn } from '../apiCall';
<<<<<<< HEAD
import { connect } from 'react-redux';
<<<<<<< HEAD
=======
import { setUser } from '../actions/index';
import * as actions from '../actions/index'
=======
import './LogIn.css'
>>>>>>> Clean up adjust movie card size and format LogIn.js
>>>>>>> Clean up adjust movie card size and format LogIn.js

class LogIn extends Component {
  constructor(props) {
    super(props);
    
    this.state ={
      email: '',
      password: '',
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    const field = e.target.name;
    const value = e.target.value;

    this.setState({ [field]: value });
  }

  loginSubmit = async (e) => {
    e.preventDefault();

    const { email, password } = this.state
    const results = await userSignIn( email, password );
    const user = await results.data;

    if (!results) {
      alert('Oops! The email and password do not match');
    } else {
      this.props.setUser(user);
      this.props.history.push('/favorites')
    }
  }

  render() {

    return (
      <div className="log-in-form">
        <form onSubmit={ this.handleSubmit }>
          <input type="text" 
                 name="email" 
                 placeholder="email address" 
                 value={ this.state.email } 
                 onChange={ this.handleChange }/>
          <input type="password" 
                 name="password" 
                 placeholder="password" 
                 value={ this.state.password } 
                 onChange={ this.handleChange }/>
          <input type="submit" />
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setUser: (user) => dispatch(setUser(user)),
});

export default withRouter(connect(null, mapDispatchToProps)(LogIn));

