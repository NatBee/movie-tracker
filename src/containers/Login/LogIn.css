input {
  width: 300px;
  padding: 10px;
  font-size: 1rem;
  display: block;
  margin: 15px auto;
}

.submit-button {
  padding: 10px;
  width: 322px;
  font-size: 1rem;
  border-radius: 0;
  margin-bottom: 15px;
}

.submit-button hover {
  border: none;
}

form {
  margin: auto;
  width: 350px;
  height: 300px;
  padding: 40px;
  background-color: black;
  opacity: 0.75;
  position: relative;
  top: 50px;
}

p{
  color: white;
}

.login-error {
  color: red;
  font-size: .75em;
  padding: 0 0 10px;
}

.acct-create {
  color: red;
  font-size: .75em;
  padding: 0 0 10px;
}