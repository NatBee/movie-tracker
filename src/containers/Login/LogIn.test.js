/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import { LogIn, mapDispatchToProps } from './LogIn';

describe.skip('LogIn', () => {
  let wrapper;

  beforeEach(() => {
    
    wrapper = shallow(<LogIn />)
  })

  it('should have a default state', () => {
    const expected = {
      email: '',
      password: '',
      errorMessage: null
    }
    console.log(wrapper.state())
  });

  it('should fetch user credentials from database', () => {
    const userCredentials = {
      "email": "tman2272@aol.com",
      "password": "password"
    }

    window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        status: 200,
        json: () => Promise.resolve(userCredentials)
      })
    })
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})