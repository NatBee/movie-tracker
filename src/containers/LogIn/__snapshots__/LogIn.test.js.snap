// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should match snapshot 1`] = `undefined`;
